src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1140,102 +1145,113 @@ private AuthenticationResult acquireTokenAfterValidation(CallbackHandler callbac
+                    // Activity needs to launch to record calling app for this
+                    // account
+                    Intent brokerIntent = mBrokerProxy.getIntentForBrokerActivity(request);
+                    if (brokerIntent != null) {
+                        try {
+                            Logger.v(TAG, "Calling activity pid:" + android.os.Process.myPid()
+                                    + " tid:" + android.os.Process.myTid() + "uid:"
+                                    + android.os.Process.myUid());
+                            activity.startActivityForResult(brokerIntent,
+                                    AuthenticationConstants.UIRequest.BROWSER_FLOW);
+                        } catch (ActivityNotFoundException e) {
+                            Logger.e(TAG, "Activity login is not found after resolving intent", "",
+                                    ADALError.DEVELOPER_ACTIVITY_IS_NOT_RESOLVED, e);
+                            callbackHandle.onError(new AuthenticationException(
+                                    ADALError.BROKER_ACTIVITY_IS_NOT_RESOLVED));
+                        }
+                    } else {
                         callbackHandle.onError(new AuthenticationException(
-                                ADALError.BROKER_ACTIVITY_IS_NOT_RESOLVED));
+                                ADALError.DEVELOPER_ACTIVITY_IS_NOT_RESOLVED));
                     }
                 } else {
+    
+                    // User does not want to launch activity
+                    String msg = "Prompt is not allowed and failed to get token:";
+                    Logger.e(TAG, msg, "", ADALError.AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED);
                     callbackHandle.onError(new AuthenticationException(
-                            ADALError.DEVELOPER_ACTIVITY_IS_NOT_RESOLVED));
+                            ADALError.AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED, msg));
                 }
+    
+                // It will start activity if callback is provided. Return null here.
+                return null;
             } else {
-
-                // User does not want to launch activity
-                String msg = "Prompt is not allowed and failed to get token:";
-                Logger.e(TAG, msg, "", ADALError.AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED);
-                callbackHandle.onError(new AuthenticationException(
-                        ADALError.AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED, msg));
+                return localFlow(callbackHandle, activity, useDialog, request);
             }
-
-            // It will start activity if callback is provided. Return null here.
+        }
+        catch(DeveloperAuthenticationException e)
+        {
+            Logger.w(TAG,
+                    e.getMessage(),
+                    "", ADALError.DEVELOPER_BROKER_PERMISSIONS_MISSING);
+            callbackHandle.onError((AuthenticationException)e);
```

Comment:
Because this exception is thrown in the if-else statement at the beginning and I don't know if there is any other way to put the catch statement in this method. Any suggestion?  @weijjia